### PR TITLE
[16.0][MIG] account_move_reconcile_forbid_cancel: Migration to 16.0

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
Issue #497

File: account_move_reconcile_forbid_cancel/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models, tools
+from odoo.exceptions import ValidationError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _get_receivable_payable_lines(self):
```
Comment (RodrigoBM) [inline account_move_reconcile_forbid_cancel/models/account_move.py]: delete this method

File: account_move_reconcile_forbid_cancel/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return self.line_ids.filtered(
+            lambda l: l.account_type in ["asset_receivable", "liability_payable"],
+        )
+
+    def button_draft(self):
+        if not self.env.context.get("skip_reconcile_forbid_cancel") and (
+            not tools.config["test_enable"]
+            or self.env.context.get("test_reconcile_forbid_cancel")
+        ):
+            rec_pay_lines = self._get_receivable_payable_lines()
```
Comment (RodrigoBM) [inline account_move_reconcile_forbid_cancel/models/account_move.py]: delete it 
```suggestion
```

File: account_move_reconcile_forbid_cancel/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            if rec_pay_lines.matched_debit_ids or rec_pay_lines.matched_credit_ids:
```
Comment (RodrigoBM) [inline account_move_reconcile_forbid_cancel/models/account_move.py]:
```suggestion
            if self._get_reconciled_amls():
```

File: account_move_reconcile_forbid_cancel/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                    _("You cannot reset to draft reconciled entries.")
+                )
+        return super().button_draft()
+
+    def button_cancel(self):
+        if not self.env.context.get("skip_reconcile_forbid_cancel") and (
+            not tools.config["test_enable"]
+            or self.env.context.get("test_reconcile_forbid_cancel")
+        ):
+            rec_pay_lines = self._get_receivable_payable_lines()
```
Comment (RodrigoBM) [inline account_move_reconcile_forbid_cancel/models/account_move.py]: delete it

File: account_move_reconcile_forbid_cancel/models/account_move.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            if rec_pay_lines.matched_debit_ids or rec_pay_lines.matched_credit_ids:
```
Comment (RodrigoBM) [inline account_move_reconcile_forbid_cancel/models/account_move.py]:
```suggestion
            if self._get_reconciled_amls():
```

**Review by RodrigoBM — CHANGES_REQUESTED** [review]
can you try it?

**Comment by ghost** [comment]
Thank you @RodrigoBM

**Review by RodrigoBM — APPROVED** [review]

**Comment by RodrigoBM** [comment]
have @adrianojprado added this translation?

https://github.com/OCA/account-reconcile/blame/16.0/account_move_base_import/i18n/pt_BR.po#L36

this is wrong, it will have to be` "%(log)s\n"`

**Comment by RodrigoBM** [comment]
> have @adrianojprado added this translation?
> 
> https://github.com/OCA/account-reconcile/blame/16.0/account_move_base_import/i18n/pt_BR.po#L36
> 
> this is wrong, it will have to be` "%(log)s\n"`

ping @adrianojprado

https://translation.odoo-community.org/translate/account-reconcile-16-0/account-reconcile-16-0-account_move_base_import/pt_BR/?checksum=1ec4daeaeafa847c

```
%(completion_date)s A movimentação da conta %(move_name)s tem %(num_imported)s/%(number_line)s linhas concluídas por %(user_name)s
%(error_msg)s
%(log)s\n
```

**Comment by RodrigoBM** [comment]
> > have @adrianojprado added this translation?
> > https://github.com/OCA/account-reconcile/blame/16.0/account_move_base_import/i18n/pt_BR.po#L36
> > this is wrong, it will have to be` "%(log)s\n"`
> 
> ping @adrianojprado
> 
> https://translation.odoo-community.org/translate/account-reconcile-16-0/account-reconcile-16-0-account_move_base_import/pt_BR/?checksum=1ec4daeaeafa847c
> 
> ```
> %(completion_date)s A movimentação da conta %(move_name)s tem %(num_imported)s/%(number_line)s linhas concluídas por %(user_name)s
> %(error_msg)s
> %(log)s\n
> ```

Thanks @adrianojprado

@ACheung-FactorLibre Can you rebase plase

**Review by RodrigoBM — APPROVED** [review]

**Review by ikedetoro — APPROVED** [review]

**Review by Gelo-fl — APPROVED** [review]
LGTM https://www.awesomescreenshot.com/video/20375290?key=b64ee17bd28e1df04f1fd0807e9f7ae2

**Pull request merged**